core/input/input.cpp

```diff
@@ -270,6 +270,17 @@ bool Input::is_mouse_button_pressed(MouseButton p_button) const {
 	return mouse_button_mask.has_flag(mouse_button_to_mask(p_button));
 }
 
+bool Input::is_buffered_key_pressed(Key p_keycode) const {
+	_THREAD_SAFE_METHOD_
+	for (Ref<InputEvent> event : buffered_events) {
+		Ref<InputEventKey> k = event;
+		if (k.is_valid() && k->get_keycode() == p_keycode && k->is_pressed()) {
+			return true;
+		}
+	}
```

Comment:
We discussed this in a PR-review meeting.
This function has the problem, that it returns `true`, whenever it finds the correct keycode. However in the case of a very long frame, multiple key-down + key-up events can happen and the current algorithm only checks if any key-down happens.
What would rather need to happen is, that the `buffered_events` need to be checked from the end for `p_keycode` and returns a boolean based on if the most recend event was a key-down or a key-up.